libcudacxx/include/cuda/__ptx/ptx_dot_variants.h

```diff
@@ -173,6 +192,35 @@ static constexpr op_xor_op_t op_xor_op{};
 static constexpr op_cas_t op_cas{};
 static constexpr op_exch_t op_exch{};
 
+template <dot_cta_group __cta_group>
+using cta_group_t   = _CUDA_VSTD::integral_constant<dot_cta_group, __cta_group>;
+using cta_group_1_t = cta_group_t<dot_cta_group::cta_group_1>;
+using cta_group_2_t = cta_group_t<dot_cta_group::cta_group_1>;
+
+static constexpr cta_group_1_t cta_group_1{};
+static constexpr cta_group_2_t cta_group_2{};
+
+template <dot_kind __kind>
+using kind_t          = _CUDA_VSTD::integral_constant<dot_kind, __kind>;
+using kind_f16_t      = kind_t<dot_kind::f16>;
+using kind_f8f6f4_t   = kind_t<dot_kind::f8f6f4>;
+using kind_i8_t       = kind_t<dot_kind::i8>;
+using kind_mxf4_t     = kind_t<dot_kind::mxf4>;
+using kind_mxf4nvf4_t = kind_t<dot_kind::mxf4nvf4>;
+using kind_mxf8f6f4_t = kind_t<dot_kind::mxf8f6f4>;
+using kind_tf32_t     = kind_t<dot_kind::tf32>;
+
+static constexpr kind_f16_t kind_f16{};
+static constexpr kind_f8f6f4_t kind_f8f6f4{};
+static constexpr kind_i8_t kind_i8{};
+static constexpr kind_mxf4_t kind_mxf4{};
+static constexpr kind_mxf4nvf4_t kind_mxf4nvf4{};
+static constexpr kind_mxf8f6f4_t kind_mxf8f6f4{};
+static constexpr kind_tf32_t kind_tf32{};
+
+template <int n>
+using n32_t = _CUDA_VSTD::integral_constant<int, n>;
```

Comment:
Question: Will this work in pre-C++17? Do we still need to care?

Reply:
I do not see anything that would not work in earlier modes, but we also do not care anymore about C++11 and C++14

Reply:
Unless ..

.. we need to backport this, which we have to in this case :(

Reply:
yeah but nothing here requires C++17?